### PR TITLE
LOJ-309 implement summary code updates

### PR DESCRIPTION
**Pull request by jasperRob**
A large scale refactor of the ProcessQuestions Process, gearing the system more towards PCM architecture to make pages a little more customisable.

Also includes some new classes to make our lives easier, and code neater.

### CODE REVIEW

File: qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java [hunk 1]
```diff
@@ -198,168 +200,141 @@ public AnswerLink(final BaseEntity source, final BaseEntity target, final Answer
...
-						try {
-							Date olddate = new SimpleDateFormat(formatString).parse(result);
-							final LocalDateTime dateTime = olddate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
-							setValueDateTime(dateTime);
-							break;
-						} catch (java.text.ParseException e) {
-							continue;
-						}
-
+				List<String> dateTimeFormats = Arrays.asList("yyyy-MM-dd'T'HH:mm:ss", "yyyy-MM-dd HH:mm",
```
Comment (varunshastry) [inline qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java]: Extract the dateTime formats to a separate file.

File: qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java [hunk 2]
```diff
@@ -198,168 +200,141 @@ public AnswerLink(final BaseEntity source, final BaseEntity target, final Answer
...
 		}
 
 		switch (className) {
-			case "life.genny.qwandaq.entity":
-				List<String> beCodeList = new CopyOnWriteArrayList<String>();
-				beCodeList.add(answer.getValue());
-				setValueBaseEntityCodeList(beCodeList);
+
+			case "life.genny.qwandaq.entity.BaseEntity":
+			case "life.genny.qwanda.entity.BaseEntity":
```
Comment (varunshastry) [inline qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java:224]: do we need both? only qwandaq variant would suffice?

File: qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java [hunk 3]
```diff
@@ -198,168 +200,141 @@ public AnswerLink(final BaseEntity source, final BaseEntity target, final Answer
...
+				}
 				break;
+
 			case "java.time.LocalTime":
 			case "LocalTime":
-				result = answer.getValue();
-				formatStrings = Arrays.asList("HH:mm", "HH:mm:ss", "HH:mm:ss.SSSZ");
-				for (String formatString : formatStrings) {
-					Date olddate;
+				List<String> timeFormats = Arrays.asList("HH:mm", "HH:mm:ss", "HH:mm:ss.SSSZ");
```
Comment (varunshastry) [inline qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java]: Extract into dateTime formats file.

**Review by varunshastry — CHANGES_REQUESTED** [review]

File: qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java (hunk 2 above)
Comment (jasperRob) [reply qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java:224]: Believe it or not, there are cases in internmatch where the old qwanda version surfaces

File: qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java (hunk 3 above)
Comment (jasperRob) [reply qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java]: Done

File: qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java (hunk 1 above)
Comment (jasperRob) [reply qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java]: Done

**Review by varunshastry — APPROVED** [review]

**Pull request merged**